### PR TITLE
OPSEXP-1742 Bump detect-secrets plugin

### PR DESCRIPTION
**Pull request by gionn**
* update detect-secrets to latest version
* remove all pragma to allow secrets AND add them to baseline as not secrets

What I've understood is:
* every time a secret detection get triggered by pre-commit hook:
  * `detect-secrets scan --baseline .secrets.baseline` to add new detections to the baseline
  * `detect-secrets audit .secrets.baseline` to mark that detection as non-secret

OPSEXP-1742

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -39,15 +39,10 @@ repos:
...
-    rev: v1.1.0
+    rev: v1.4.0
     hooks:
       - id: detect-secrets
-        name: detect generic secrets
-        args:
-          - --baseline
-          - .secrets.baseline
-          - --exclude-secrets
-          - (dummy|admin|mp6yc0UD9e|alfresco)
```
Comment (alxgomz) [inline .pre-commit-config.yaml:50]: I would keep these in order to not be bothered IF some secret really needs to be added. That gives people an option to add secret without having to deal with secrets baseline (something to document)

File: .secrets.baseline [hunk 2]
```diff
@@ -100,33 +109,225 @@
     },
     {
       "path": "detect_secrets.filters.heuristic.is_templated_secret"
-    },
-    {
-      "path": "detect_secrets.filters.regex.should_exclude_secret",
-      "pattern": [
-        "(admin|mp6yc0UD9e|alfresco)"
```
Comment (alxgomz) [inline .secrets.baseline:107]: maybe using this is even better compared to pre-commit config arg

File: .pre-commit-config.yaml (hunk 1 above)
Comment (gionn) [reply .pre-commit-config.yaml:50]: I've discovered that if you update the baseline providing exclusions they become persistent into the baseline itself

File: .secrets.baseline (hunk 2 above)
Comment (alxgomz) [reply .secrets.baseline:107]: I'd rather we: 
* keep it for convenience
* document theese password (at least admin & alfresco) as dummy ones

**Review by alxgomz — CHANGES_REQUESTED** [review]
let's keep the dummy password in the baseline config

**Review by alxgomz — APPROVED** [review]

**Pull request merged**